deps.ffmpeg/10-zlib.ps1

```diff
@@ -0,0 +1,87 @@
+param(
+    [string] $Name = 'zlib',
+    [string] $Version = '1.2.13',
+    [string] $Uri = 'https://github.com/madler/zlib.git',
+    [string] $Hash = "04f42ceca40f73e2978b50e93806c2a18c1281fc",
+    [array] $Patches = @(
+        @{
+            PatchFile = "${PSScriptRoot}/patches/zlib/0001-fix-unistd-detection.patch"
+            HashSum = "2114ff9ebfc79765019353b06915a09f4dc4802ce722d2df6e640a59666dd875"
+        }
+    )
+)
+
+function Setup {
+    Setup-Dependency -Uri $Uri -Hash $Hash -DestinationPath $Path
+}
+
+function Clean {
+    Set-Location $Path
+    if ( Test-Path "build_${Target}" ) {
+        Log-Information "Clean build directory (${Target})"
+        Remove-Item -Path "build_${Target}" -Recurse -Force
+    }
+}
+
+function Patch {
+    Log-Information "Patch (${Target})"
+    Set-Location $Path
+
+    $Patches | ForEach-Object {
+        $Params = $_
+        Safe-Patch @Params
+    }
+}
+
+function Configure {
+    Log-Information "Configure (${Target})"
+    Set-Location $Path
+
+    $OnOff = @('OFF', 'ON')
+    $Options = @(
+        $CmakeOptions
+        '-DZ_HAVE_UNISTD_H:BOOL=OFF'
+    )
```

Comment:
I suppose we don't need these if we aren't building this shared:
```
-DCMAKE_SHARED_LIBRARY_PREFIX=""
-DCMAKE_SHARED_LIBRARY_PREFIX_C=""
```

Though I don't recall why we build shared presently and why this PR no longer does. If we're not building this shared, we also don't need the `$OnOff` variable.

Reply:
In general we don't seem to need it because when using the MSVC generator, their CMake detects that we're on windows and it omits the `lib` prefix by itself.